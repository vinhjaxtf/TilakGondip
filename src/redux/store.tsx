import { createStore, applyMiddleware, compose, combineReducers } from "redux";
import thunk from "redux-thunk";
import { book } from "./reducers/book";
import { manager } from "./reducers/manager";
import { progressPanel } from "./reducers/progressPanel";
import { reader } from "./reducers/reader";
import { viewArea } from "./reducers/viewArea";
import { sidebar } from "./reducers/sidebar";
import { backupPage } from "./reducers/backupPage";
import BookModel from "../model/Book";
import NoteModel from "../model/Note";
import BookmarkModel from "../model/Bookmark";
const rootReducer = combineReducers({
  book,
  manager,
  reader,
  progressPanel,
  viewArea,
  sidebar,
  backupPage,
});
const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(thunk),
    (window as any).devToolsExtension
      ? (window as any).devToolsExtension()
      : (f: any) => f
  )
);
export default store;
export type stateType = {
  manager: {
    books: BookModel[];
    epubs: any[];
    covers: { key: string; url: string }[];
    searchResults: number[];
    isSearch: boolean;
    isSort: boolean;
    isFirst: string;
    isList: string;
    isSortDisplay: boolean;
    sortCode: { sort: number; order: number };
    isMessage: boolean;
    message: string;
  };
  book: {
    isOpenEditDialog: boolean;
    isOpenDeleteDialog: boolean;
    isOpenAddDialog: boolean;
    isOpenActionDialog: boolean;
    isReading: boolean;
    currentBook: BookModel;
    currentEpub: any;
  };
  backupPage: {
    isBackup: boolean;
    isOpenTokenDialog: boolean;
  };
  progressPanel: {
    percentage: number;
    locations: any[];
  };
  reader: {
    bookmarks: BookmarkModel[];
    notes: NoteModel[];
    digests: NoteModel[];
    locations: any[];
    color: number;
    chapters: any[];
    flattenChapters: any;
    noteKey: string;
    originalText: string;
  };
  sidebar: {
    mode: string;
    shelfIndex: number;
  };
  viewArea: {
    selection: string;
    menuMode: string;
    isOpenMenu: boolean;
    isChangeDirection: boolean;
    isShowBookmark: boolean;
  };
};
