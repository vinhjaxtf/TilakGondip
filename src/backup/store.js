import { createStore, applyMiddleware, compose, combineReducers } from "redux";
import thunk from "redux-thunk";
import { settingPanel } from "./redux/settingPanel.redux";
import { book } from "./redux/book.redux";
import { manager } from "./redux/manager.redux";
import { progressPanel } from "./redux/progressPanel.redux";
import { reader } from "./redux/reader.redux";
import { viewArea } from "./redux/viewArea.redux";
import { sidebar } from "./redux/sidebar.redux";
import { backupPage } from "./redux/backupPage.redux";
const rootReducer = combineReducers({
  settingPanel,
  book,
  manager,
  reader,
  progressPanel,
  viewArea,
  sidebar,
  backupPage
});
// console.log(manager, rootReducer);
const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(thunk),
    window.devToolsExtension ? window.devToolsExtension() : f => f
  )
);
export default store;
