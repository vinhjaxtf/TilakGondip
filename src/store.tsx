import { createStore, applyMiddleware, compose, combineReducers } from "redux";
import thunk from "redux-thunk";
import { book } from "./redux/book.redux";
import { manager } from "./redux/manager.redux";
import { progressPanel } from "./redux/progressPanel.redux";
import { reader } from "./redux/reader.redux";
import { viewArea } from "./redux/viewArea.redux";
import { sidebar } from "./redux/sidebar.redux";
import { backupPage } from "./redux/backupPage.redux";
import BookModel from "./model/Book";
import NoteModel from "./model/Note";
import DigestModel from "./model/Digest";
import HighligherModel from "./model/Highlighter";
import BookmarkModel from "./model/Bookmark";
const rootReducer = combineReducers({
  book,
  manager,
  reader,
  progressPanel,
  viewArea,
  sidebar,
  backupPage,
});
const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(thunk),
    (window as any).devToolsExtension
      ? (window as any).devToolsExtension()
      : (f: any) => f
  )
);
export default store;
export type stateType = {
  manager: {
    books: BookModel[];
    epubs: any[];
    covers: { key: string; url: string }[];
    searchBooks: number[];
    isSearch: boolean;
    isSort: boolean;
    isList: string;
    isSortDisplay: boolean;
    sortCode: { sort: number; order: number };
    isMessage: boolean;
    message: string;
  };
  book: {
    isOpenEditDialog: boolean;
    isOpenDeleteDialog: boolean;
    isOpenAddDialog: boolean;
    isReading: boolean;
    currentBook: BookModel;
    currentEpub: any;
  };
  backupPage: {
    isBackup: boolean;
  };
  progressPanel: {
    percentage: number;
    locations: any[];
  };
  reader: {
    bookmarks: BookmarkModel[];
    notes: NoteModel[];
    digests: DigestModel[];
    locations: any[];
    chapters: any[];
    highlighters: HighligherModel[];
    isSingle: string;
  };
  sidebar: {
    mode: string;
    shelfIndex: number;
  };
  viewArea: {
    selection: string;
    highlighters: HighligherModel[];
  };
};
