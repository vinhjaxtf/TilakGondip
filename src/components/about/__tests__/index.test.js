import React from "react";
import { render } from "@testing-library/react";

import { Provider } from "react-redux";
import About from "../";
import store from "../../../redux/store";

describe("<About />", () => {
  it("should render about info", () => {
    const { container } = render(
      <Provider store={store}>
        <About />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
