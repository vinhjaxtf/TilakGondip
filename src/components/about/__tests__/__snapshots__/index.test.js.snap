// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render about info 1`] = `
<div
  class="about-container"
>
  <div
    class="about-icon-container"
  >
    <span
      class="icon-github about-icon"
    />
    <span
      class="icon-contact about-icon"
    />
    <span
      class="icon-donate about-icon"
    />
  </div>
</div>
`;
